R/DiseasyModel.R

```diff
@@ -0,0 +1,200 @@
+#' @title Base module for diseasy model templates
+#'
+#' @description
+#'   The `DiseasyModel` module implements common functionality that all models have available beyond that provided by
+#'   `DiseasyBaseModule`.
+#'   Most notably, the model module facilitates:
+#'   * Module interfaces:
+#'     The module contains the functional modules via its active bindings:
+#'     * `$activity`: `DiseasyActivity`
+#'     * `$observables`: `DiseasyObservables`
+#'     * `$season`: `DiseasySeason`
+#'     Configured instances of these modules can be provided during initialisation. If not, default instances are
+#'     created.
+#'   * Model interface:
+#'     The module defines the functions `$get_results()`, `$get_training_data()` and the `$parameters` binding.
+#'     These functions define the "API" of the models and ensure the models can take part in the ensemble.
+#' @examples
+#'   # Normally, you would not want to create this module directly, but it is possible.
+#'   Model_module <- DiseasyModel$new()
+#'
+#'   rm(Model_module)
+#' @return
+#'   A new instance of the `DiseasyModel` [R6][R6::R6Class] class.
+#' @export
+#' @seealso [lgr][lgr::lgr]
+DiseasyModel <- R6::R6Class(                                                                                            # nolint: object_name_linter
+  classname = "DiseasyModel",
+  inherit = DiseasyBaseModule,
+
+  public = list(
+
+    #' @description
+    #'   Creates a new instance of the `DiseasyModel` [R6][R6::R6Class] class.
+    #'   This module is typically not constructed directly but rather through `DiseasyModel*` classes
+    #' @param activity,observables,season (`boolean` or `R6::R6Class instance`)\cr
+    #'   If a boolean is given, it dictates whether to load a new instance module of this class\cr
+    #'   If an instance of the module is provided instead, this instance is cloned to the new `DiseasyModel` instance\cr
```

Comment:
Is it clear what 'cloned' means for the user? I suggest that we at some point look into this in a vignette.
It is not clear to me, what happens when the value is FALSE ... implicitly it states that it will not load a new instance. But in the description above (line 12) it states that default instances are created if not provided.

Reply:
I have made some changes to the wording. Let me know if it is more clear now